.pre-commit-config.yaml

```diff
@@ -13,9 +13,19 @@ repos:
           - --strict
 
   - repo: https://github.com/python-poetry/poetry
-    rev: 1.5.0
+    rev: 1.8.3
     hooks:
       - id: poetry-check
+        args:
```

Comment:
What does this hook do, does it check if the lock file is up to date or does it also update and commit it ?

Reply:
As far as I can tell it runs [`poetry check --lock`](https://python-poetry.org/docs/cli/#check). The documentation is not super clear on what this does, I assume there is an issue in the documentation, because the behavior does not match the documentation.

Anyway what actually happens in this hook is that it checks that:
1. the content of `pyproject.toml` is valid (from the point of view of Poetry)
2. there is a `poetry.lock` file
3. the content of `poetry.lock` is in sync with `pyproject.toml`

The next hook runs [`poetry lock --no-update`](https://python-poetry.org/docs/cli/#lock), which does syncs the lock file. The hook will fail if the hook changes the file, this is important to have on the CI side, for the case when someone commits without running the hooks locally.

Then the third hook does the export to `requirements.txt`. The hook will fail if the hook changes the file, this is important to have on the CI side, for the case when someone commits without running the hooks locally.

There is some redundancy between the first two steps, and I assume we can simplify over time, depending on how things go.